src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -2731,13 +2733,730 @@ def cli_cosmosdb_managed_cassandra_datacenter_update(client,
+    restorable_database_accounts = restorable_database_accounts_client.list()
+    restorable_database_accounts_list = list(restorable_database_accounts)
+    restorable_database_account = None
+
+    if restore_timestamp is not None:
+        restore_timestamp_datetime_utc = _convert_to_utc_timestamp(restore_timestamp)
+        for account in restorable_database_accounts_list:
+            if account.account_name == account_name:
+                if account.deletion_time is not None:
+                    if account.deletion_time >= restore_timestamp_datetime_utc >= account.creation_time:
+                        raise CLIError("Cannot perform inaccount restore on a deleted database account {}".format(account_name))
+                else:
+                    if restore_timestamp_datetime_utc >= account.creation_time:
+                        restorable_database_account = account
+                        break
+
+        if restorable_database_account is None:
+            raise CLIError("Cannot find a database account with name {} that is online at {}".format(account_name, restore_timestamp))
+    else:
+        latest_account_to_restore = None
+        for account in restorable_database_accounts_list:
+            if account.account_name == account_name:
+                if latest_account_to_restore is None:
+                    latest_account_to_restore = account
+
+                if account.creation_time > latest_account_to_restore.creation_time:
+                    if account.deletion_time is None:
+                        latest_account_to_restore = account
+
+        restorable_database_account = latest_account_to_restore
+
+        if restorable_database_account is None:
+            raise CLIError("Cannot find a database account with name {} that is online".format(account_name))
+
+        regex = "[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"
+        account_instance_id = re.findall(regex, restorable_database_account.id)[1]
+        latest_restore_time = datetime.datetime.utcfromtimestamp(0)
+        try:
+            from azure.cli.command_modules.cosmosdb._client_factory import cf_restorable_sql_databases
+            restorable_databases_client = cf_restorable_sql_databases(cmd.cli_ctx, [])
+            restorable_databases = restorable_databases_client.list(
+                restorable_database_account.location,
+                account_instance_id)
+
+            for restorable_database in restorable_databases:
+                database = restorable_database.resource
+                event_timestamp = datetime.datetime.strptime(database.event_timestamp, "%Y-%m-%dT%H:%M:%SZ")
+                if database.operation_type == "Delete" and latest_restore_time < event_timestamp and database.owner_id == database_name:
+                    latest_restore_time = event_timestamp
+
+            restore_time = latest_restore_time + datetime.timedelta(seconds=-1)
```

Comment:
nit: add a comment here on why we are subtracting 1 second.

Reply:
updated as per suggestion